tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,182 @@
+---
+title: "NServiceBus 101 Lesson 2: Sending a command"
+---
+
+Sending and receiving messages is at the core of any NServiceBus system. Durable messages passed between processes allow reliable communication between those processes, even if one of them is temporarily unavailable. In this lesson we'll show how to send and process a message.
+
+
+## Objectives
+
+By the end of this lesson, you will have learned how to:
+
+ * Define messages
+ * Define message handlers
+ * Send and receive a message locally
+ * Use NServiceBus's logging capabilities
+
+
+## What is a message?
+
+A [**message**](/nservicebus/messaging/messages-events-commands.md) is a collection of data sent via one-way communication between two endpoints. In NServiceBus, we can define a message via a simple class.
+
+In this lesson, we'll focus on the simpler type of message: [commands](/nservicebus/messaging/messages-events-commands.md#command). In [Lesson 4: Publishing events](../lesson-4/) we'll expand to look at events as well.
+
+Defining a command is pretty easy. We just create a class and mark it with the `ICommand` marker interface.
+
+snippet:Command
+
+The marker interface has no implementation, but lets NServiceBus know that the class is a command so that it can build up some metadata about the message type when the endpoint starts up. Any properties you create within the message constitute the message data.
+
+The name of the command class is also important. A command is an order to do something, so it should be named in the [imperative tense](https://en.wikipedia.org/wiki/Imperative_mood). `PlaceOrder`, `UpdateCustomerStatus`, and `ChargeCreditCard` are all great names for commands, because they are phrased as a command and are very specific. You can easily guess exactly what each of those messages will do. `UserMessage`, on the other hand, is not a good example. It is not in the imperative, and it's not very specific either. Another developer should know exactly what a command's purpose is just by reading the name.
```

Comment:
> `UpdateCustomerStatus`

That kind of can easily become "UpdatePropertyX", maybe "MarkCustomerAsGold" or something like that (more business-oriented).